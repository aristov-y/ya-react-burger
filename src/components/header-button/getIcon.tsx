import {
  BurgerIcon,
  ListIcon,
  ProfileIcon
} from '@ya.praktikum/react-developer-burger-ui-components';
import React from 'react';

function getIcon(iconType: 'BURGER' | 'LIST' | 'PROFILE', active: boolean = false) {
  const type = active ? 'primary' : 'secondary';
  switch (iconType) {
    case 'BURGER':
      return (<BurgerIcon type={type} />);
    case 'LIST':
      return (<ListIcon type={type} />);
    case 'PROFILE':
      return (<ProfileIcon type={type} />);
    default:
      return null as never;
  }
}

export default getIcon;
