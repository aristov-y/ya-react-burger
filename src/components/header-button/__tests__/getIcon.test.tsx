import { create, act, ReactTestRenderer } from 'react-test-renderer';
import getIcon from '../getIcon';
import { BurgerIcon } from '@ya.praktikum/react-developer-burger-ui-components';

describe('getIcon', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(getIcon('BURGER'));
    });

    expect(component.toJSON()).toMatchSnapshot();

    const instance = component.root;

    let burger = instance.findByType(BurgerIcon);
    expect(burger.props.type).toBe('secondary');

    act(() => {
      component.update(getIcon('BURGER', true));
    });

    expect(component.toJSON()).toMatchSnapshot();

    burger = instance.findByType(BurgerIcon);
    expect(burger.props.type).toBe('primary');

    act(() => {
      component.update(getIcon('LIST'));
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(getIcon('PROFILE'));
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(getIcon(undefined));
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
