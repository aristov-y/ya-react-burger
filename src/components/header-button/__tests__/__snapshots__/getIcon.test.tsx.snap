// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIcon should render 1`] = `"BurgerIcon"`;

exports[`getIcon should render 2`] = `"BurgerIcon"`;

exports[`getIcon should render 3`] = `"ListIcon"`;

exports[`getIcon should render 4`] = `"ProfileIcon"`;

exports[`getIcon should render 5`] = `null`;
